addons/web/static/src/views/list/list_renderer.scss

```
@@ -67,6 +71,10 @@
             }
         }
 
+        @include media-breakpoint-down(sm) {
+            width: auto;
```

Add a note explaining the desired behavior, eg:
```suggestion
            // On mobile, allow the table to extend to it's natural size.
            width: auto;
```